opening_hours/utils/opening_hours_client.py

```diff
@@ -164,12 +168,20 @@ def get_resource_periods(self, resource) -> List[Period]:
         return self.periods.get(resource)
 
     @datetime_args_to_default_timezone
-    def is_resource_open(
+    def is_resource_open_for_reservations(
         self, resource: str, start_time: datetime.datetime, end_time: datetime.datetime
     ) -> bool:
         times = self.get_opening_hours_for_resource(resource, start_time.date())
-        for time in times:
-            if time.start_time <= start_time and time.end_time >= end_time:
+        for time in [
+            time
+            for time in times
+            if ResourceState(time.resource_state) in ResourceState.reservable_states()
+        ]:
+            if (
+                time.start_time <= start_time
+                and (time.end_time >= end_time or time.end_time_on_next_day)
```

Comment:
Hmm, I wonder if the first two lines should be in parenthesis 🤔 The first and last condition are both checking `time.start_time` but on the latter it is a falsy check and not comparison to some other value. So something like this:
```
(
    time.start_time <= start_time
    and (time.end_time >= end_time or time.end_time_on_next_day)
)
or (not time.start_time and not time.end_time)
```

Or maybe even other way round just to make the falsy check first 🤔

Reply:
nice ones :+1:  I'll refactor

Reply:
Added the falsy check as variable to be more readable